smapi.go

```diff
@@ -611,6 +611,83 @@ func (h *Client) Delete(ctx context.Context, url string, auth bool) (*http.Respo
 	return h.do(ctx, url, http.MethodDelete, auth, nil, nil)
 }
 
+// Put is a utility method to send a PUT request to the SM API.
+//
+// The `url` argument specifies the additional URL path of the request (minus
+// the base, which is part of the client). `auth` specifies whether or not to
+// include authorization headers. `body` specifies the request body, and
+// `headers` specifies the request headers.
+func (h *Client) Put(ctx context.Context, url string, auth bool, body io.Reader, headers http.Header) (*http.Response, error) {
```

Comment:
nit: Both the `Post()` and `do()` methods accept the headers field before the body. Should we change that for coherency?
```suggestion
func (h *Client) Put(ctx context.Context, url string, auth bool, headers http.Header, body io.Reader) (*http.Response, error) {
```

If we change this we'll have to change also [this line](https://github.com/grafana/synthetic-monitoring-api-go-client/blob/a3225b520d76f95199f6f2430313306cc8e691ae/smapi.go#L642).